website/addons/box/serializer.py

```diff
@@ -0,0 +1,121 @@
+from website.addons.base.serializer import OAuthAddonSerializer
+
+from website.util import api_url_for, web_url_for
+
+from box.client import BoxClient, BoxClientException
+
+
+class BoxSerializer(OAuthAddonSerializer):
+
+    def credentials_owner(self, user_settings=None):
+        return user_settings.owner or self.user_settings.owner
+
+    def serialize_folder(self, folder):
+        """Serializes metadata to a dict with the display name and path
+        of the folder.
+        """
+        # if path is root
+        if folder['path'] == '' or folder['path'] == '/':
+            name = '/ (Full Box)'
+        else:
+            name = 'Box' + folder['path']
+        return {
+            'data': folder,
+            'kind': 'folder',
+            'name': name,
+            'path': folder['path'],
+        }
+
+    @property
+    def user_is_owner(self):
+        if self.user_settings is None or self.node_settings is None:
+            return False
+
+        user_accounts = self.user_settings.external_accounts
+        return bool(
+            (
+                self.node_settings.has_auth and
+                (self.node_settings.external_account in user_accounts)
+            ) or len(user_accounts)
+        )
+
+    @property
+    def serialized_urls(self):
+        node = self.node_settings.owner
+
+        return {
+            'auth': api_url_for('oauth_connect',
+                                service_name='box'),
+            'settings': web_url_for('user_addons'),
+            'importAuth': node.api_url_for('box_add_user_auth'),
+            'files': node.web_url_for('collect_file_trees'),
+            'folders': node.api_url_for('box_folder_list'),
+            'config': node.api_url_for('box_set_config'),
+            'emails': node.api_url_for('box_get_share_emails'),
+            'share': 'https://app.box.com/files/0/f/{0}'.format(self.node_settings.folder_id),
+            'deauthorize': node.api_url_for('box_remove_user_auth'),
+            'accounts': node.api_url_for('box_get_user_settings'),
+        }
+
+    @property
+    def addon_serialized_urls(self):
+        node = self.node_settings.owner
```

Comment:
Is this property necessary? It looks almost identical to `BoxSerializer.serialized_urls`.